engine/buildkit/client.go

```diff
@@ -62,24 +63,24 @@ type Opts struct {
 	MainClientCallerID string
 	DNSConfig          *oci.DNSConfig
 	Frontends          map[string]bkfrontend.Frontend
+
+	ExecMetadataMu *sync.Mutex
+	execMetadata   map[digest.Digest]ContainerExecUncachedMetadata
+	RefsMu         *sync.Mutex
+	refs           map[*ref]struct{}
```

Comment:
Why do the mutexes here need to be public? I'm guessing the rationale was so that we avoid duplicating the mutexes in the `Opts`, but we avoid that anyways since we're never copying the `Opts` struct (and govet's `copylocks` would prevent us from doing this).

Also, naming-wise, it doesn't seem right to bundle these into `Opts`, since these are modified (and don't follow the `Opts` convention). We should split these out into a separate shared data struct - I can imagine these expanding in scope later.

Reply:
Yeah at one point `Opts` wasn't passed around as a pointer, which necessitated this but you're right now that I changed `Opts` to a pointer these mutexes can be de-pointered and made private. 

Will split out to a separate struct for extra clarity too.